sklearn/cluster/_kmeans.py

```diff
@@ -2138,9 +2138,18 @@ def fit(self, X, y=None, sample_weight=None):
             # Perform the iterative optimization until convergence
             for i in range(n_steps):
                 # Sample a minibatch from the full dataset
-                minibatch_indices = random_state.randint(0, n_samples, self._batch_size)
+                minibatch_indices = random_state.choice(
+                    n_samples,
+                    self._batch_size,
+                    p=sample_weight / np.sum(sample_weight),
```

Comment:
For performance reasons, we should compute `sw / sw.sum()` outside of the loop since it's always the same.